test/org/sagebionetworks/bridge/BridgeUtilsTest.java

```diff
@@ -53,14 +55,43 @@ public void commaListToSet() {
         Set<String> set = BridgeUtils.commaListToSet("a, b , c");
         assertEquals(Sets.newHashSet("a","b","c"), set);
         
+        set = BridgeUtils.commaListToSet("a,b,c");
+        assertEquals(Sets.newHashSet("a","b","c"), set);
+        
         set = BridgeUtils.commaListToSet("");
         assertEquals(Sets.newHashSet(), set);
         
         set = BridgeUtils.commaListToSet(null);
-        assertEquals(Sets.newHashSet(), set);
+        assertNotNull(set);
         
         set = BridgeUtils.commaListToSet(" a");
         assertEquals(Sets.newHashSet("a"), set);
+        
+        // Does not produce a null value.
+        set = BridgeUtils.commaListToSet("a,,b");
+        assertEquals(Sets.newHashSet("a","b"), set);
+    }
+    
+    @Test
+    public void setToCommaList() {
+        Set<String> set = Sets.newHashSet("a", null, "", "b");
+        
+        assertEquals("a,b", BridgeUtils.setToCommaList(set));
+        assertNull(BridgeUtils.setToCommaList(null));
+        assertNull(BridgeUtils.setToCommaList(Sets.newHashSet()));
     }
 
+    @Test(expected = UnsupportedOperationException.class)
+    public void nullsafeImmutableSet() {
+        assertNotNull(BridgeUtils.nullSafeImmutableSet(null));
+        assertNotNull(BridgeUtils.nullSafeImmutableSet(Sets.newHashSet("A")));
+        
+        // This should throw an UnsupportedOperationException
+        Set<String> set = BridgeUtils.nullSafeImmutableSet(Sets.newHashSet("A"));
+        set.add("B");
+        
+        // nulls are removed. They have to be to create ImmutableSet
+        assertEquals(Sets.newHashSet("A"), BridgeUtils.nullSafeImmutableSet(Sets.newHashSet(null, "A")));
```

Comment:
This assert is dead code, since the previous line will throw. We should separate this out into a separate test.